osu.Game/OsuGameBase.cs

```diff
@@ -344,27 +344,34 @@ private void load(ReadableKeyCombinationProvider keyCombinationProvider)
 
             // add api components to hierarchy.
             if (API is APIAccess apiAccess)
-                AddInternal(apiAccess);
+                Add(apiAccess);
 
-            AddInternal(spectatorClient);
-            AddInternal(MultiplayerClient);
-            AddInternal(metadataClient);
-            AddInternal(soloStatisticsWatcher);
+            Add(spectatorClient);
+            Add(MultiplayerClient);
+            Add(metadataClient);
+            Add(soloStatisticsWatcher);
 
-            AddInternal(rulesetConfigCache);
+            Add(rulesetConfigCache);
+
+            PreviewTrackManager previewTrackManager;
+            dependencies.Cache(previewTrackManager = new PreviewTrackManager(BeatmapManager.BeatmapTrackStore));
+            Add(previewTrackManager);
+
+            Add(MusicController = new MusicController());
+            dependencies.CacheAs(MusicController);
+
+            MusicController.TrackChanged += onTrackChanged;
+            Add(beatmapClock);
```

Comment:
Why is this reordered?

Reply:
Because no component that's cached should never be added and loaded after game content, that's the main point of this PR.